dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/AsyncRpcResult.java

```diff
@@ -219,6 +219,8 @@ public Result whenCompleteWithContext(BiConsumer<Result, Throwable> fn) {
         if (setFutureWhenSync || ((RpcInvocation) invocation).getInvokeMode() != InvokeMode.SYNC) {
             // Necessary! update future in context, see https://github.com/apache/dubbo/issues/9461
             RpcContext.getServiceContext().setFuture(new FutureAdapter<>(this.responseFuture));
+        } else {
+            FutureContext.getContext().clearFuture();
```

Comment:
Clear here might not a good idea. It would be better to skip set in advance.

Reply:
i will do this